complex/src/lib.rs

```diff
@@ -150,6 +152,48 @@ impl<T: Clone + Float> Complex<T> {
         let (r, theta) = self.to_polar();
         Complex::from_polar(&(r.sqrt()), &(theta/two))
     }
+    
+    /// Raises `self` to a floating point power.
+    #[inline]
+    pub fn powf(&self, exp: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        Complex::from_polar(&r.powf(exp), &(theta*exp))
+    }
+    
+    /// Returns the logarithm of `self` with respect to an arbitrary base.
+    #[inline]
+    pub fn log(&self, base: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        Complex::new(r.log(base), theta * T::one() / base.ln())
+    }
+    
+    /// Raises `self` to a complex power.
+    #[inline]
+    pub fn pow(&self, exp: Complex<T>) -> Complex<T> {
```

Comment:
Hmm, I do prefer `powc` for this name.  FWIW #106 also started this, and arrived at the same formula as you.  But the implementation and testing was lacking, and the author hasn't responded since, so we can just go with yours.